src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/TextReasoningContent.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents text reasoning content in a chat.
+/// </summary>
+/// <remarks>
+/// <see cref="TextReasoningContent"/> is distinct from <see cref="TextContent"/>. <see cref="TextReasoningContent"/>
+/// represents "thinking" or "reasoning" performed by the model and is distinct from the actual output text from
+/// the model, which is represented by <see cref="TextContent"/>. Neither types derives from the other.
+/// </remarks>
+[DebuggerDisplay("{DebuggerDisplay,nq}")]
+public sealed class TextReasoningContent : AIContent
```

Comment:
Given that you're going for a bespoke derived type over it being a boolean on the base, I take it we believe reasoning is always going to be in text form?